src/libvirtApi/domain.js

```diff
@@ -677,81 +629,66 @@ export function domainForceReboot({
-                // but undefine either way -- if we  can't get info about the VM, don't show it
-                store.dispatch(undefineVm({ connectionName, id: objPath }));
-            });
+    const props = {};
+
+    try {
+        const [domainXML] = await call(connectionName, objPath, 'org.libvirt.Domain', 'GetXMLDesc', [Enum.VIR_DOMAIN_XML_SECURE], { timeout, type: 'u' });
+        const [domInactiveXml] = await call(connectionName, objPath, 'org.libvirt.Domain', 'GetXMLDesc', [Enum.VIR_DOMAIN_XML_SECURE | Enum.VIR_DOMAIN_XML_INACTIVE], { timeout, type: 'u' });
+        props.inactiveXML = parseDomainDumpxml(connectionName, domInactiveXml, objPath);
+        props.connectionName = connectionName;
+        props.id = objPath;
+
+        const [returnProps] = await call(connectionName, objPath, "org.freedesktop.DBus.Properties", "GetAll", ["org.libvirt.Domain"], { timeout, type: 's' });
+
+        /* Sometimes not all properties are returned, for example when some domain got deleted while part
+        * of the properties got fetched from libvirt. Make sure that there is check before reading the attributes.
+        */
+        if ("Name" in returnProps)
+            props.name = returnProps.Name.v.v;
+        if ("Persistent" in returnProps)
+            props.persistent = returnProps.Persistent.v.v;
+        if ("Autostart" in returnProps)
+            props.autostart = returnProps.Autostart.v.v;
+        props.ui = resolveUiState(props.name, props.connectionName);
+
+        const dumpxmlParams = parseDomainDumpxml(connectionName, domainXML, objPath);
+        Object.assign(props, dumpxmlParams);
+
+        const domCaps = await domainGetCapabilities({ connectionName, arch: dumpxmlParams.arch, model: dumpxmlParams.emulatedMachine });
+        props.capabilities = {
+            loaderElems: getDomainCapLoader(domCaps),
+            maxVcpu: getDomainCapMaxVCPU(domCaps),
+            cpuModels: getDomainCapCPUCustomModels(domCaps),
+            cpuHostModel: getDomainCapCPUHostModel(domCaps),
+            supportedDiskBusTypes: getDomainCapDiskBusTypes(domCaps),
+        };
+
+        const [state] = await call(connectionName, objPath, 'org.libvirt.Domain', 'GetState', [0], { timeout, type: 'u' });
+        const stateStr = DOMAINSTATE[state[0]];
+
+        if (!domainIsRunning(stateStr))
+            props.actualTimeInMs = -1;
+
+        logDebug(`${props.name}.GET_VM(${objPath}, ${connectionName}): update props ${JSON.stringify(props)}`);
+
+        await store.dispatch(updateOrAddVm({ state: stateStr, ...props }));
+
+        clearVmUiState(props.name, connectionName);
+
+        await snapshotGetAll({ connectionName, domainPath: objPath });
+    } catch (ex) {
+        // "not found" is an expected error, as this runs on Stopped/Undefined events; so be quiet about these
+        if (ex.message.startsWith("Domain not found"))
+            logDebug(`GET_VM: domain ${connectionName} ${objPath} went away, undefining: ${ex.toString()}`);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1397-20240122-152937-9774631b-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)